src/transformations/utils/build-super-calls-for-chain.ts

```diff
@@ -83,20 +84,59 @@ export function buildSuperCallsForChain(
   const linearizedCtorCalls: string[] = [];
   // this is where we store the parents of uninitialized parents if any
   const notInitializable = new Set<number>();
+  const argsValues: Record<number, Expression> = {};
+  const parentArgsValues: Record<number, Expression[]> = {};
   // Remove uninitialized parents's parents from linearization, and erase if they already are linearized
   for (const parentNode of chain) {
     if (parentNode !== contractNode) {
-      // step 1 check if initializable
       const args = ctorCalls[parentNode.id]?.call?.arguments;
-      // To be initializable means that said parent has all the variables needed for the constuctor
-      const initializable = args !== undefined || isImplicitlyConstructed(parentNode);
-      if (!initializable) {
-        // If a parent is not initializable, we assume its parents aren't initializable either,
-        // because we may not have their constructor arguments.
-        // The user will invoke them anyway in the chained initializer of this parent, which
-        // will have to be manually called.
-        for (const parent of parentNode.linearizedBaseContracts) {
-          notInitializable.add(parent);
+      const parameters = getConstructor(parentNode)?.parameters.parameters;
+      // Check if any argument is literal
+      if (args && parameters) {
+        parentArgsValues[parentNode.id] = [];
+        for (let arg of args) {
+          // Need to use index since the arg does not contain a referenceDeclaration id to match with the parameter id.
+          // Parameters and arguments are in the same order so the index works for both.
+          const index = args.indexOf(arg);
+          const argId = parameters[index].id;
+          if (arg.nodeType === 'Literal') {
+            //save it in case other parent argument uses the id of the literal value as referencedDeclaration
+            argsValues[argId] = arg;
+          } else if (arg.nodeType === 'Identifier') {
+            if (arg.referencedDeclaration && argsValues[arg.referencedDeclaration]) {
```

Comment:
It's generally not a good idea to use a value of `number | undefined` as a boolean expression, because 0 is a falsy value. So if the `referencedDeclaration` was 0 we would ignore it (it's interpreted as `false`) but it's not what we want. We need to check for undefined explicitly.